src/Server.php

```diff
@@ -275,19 +275,22 @@ function ($error) use ($that, $conn, $request) {
     /** @internal */
     public function writeError(ConnectionInterface $conn, $code, ServerRequestInterface $request = null)
     {
-        $message = 'Error ' . $code;
-        if (isset(ResponseCodes::$statusTexts[$code])) {
-            $message .= ': ' . ResponseCodes::$statusTexts[$code];
-        }
-
         $response = new Response(
             $code,
             array(
                 'Content-Type' => 'text/plain'
             ),
-            $message
+            'Error ' . $code
         );
 
+        // append reason phrase to response body if known
+        $reason = $response->getReasonPhrase();
+        if ($reason !== '') {
+            $body = $response->getBody();
+            $body->seek(0, SEEK_END);
```

Comment:
This makes sure it seeks to the end so that we *append* the reason phrase to the existing body.